allennlp/training/metrics.py

```diff
@@ -0,0 +1,190 @@
+from typing import Optional, Tuple, Union
+
+from overrides import overrides
+import torch
+
+from allennlp.common.registrable import Registrable
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+
+
+class Metric(Registrable):
+    """
+    A very general abstract class representing a metric which can be
+    accumulated.
+    """
+    def __call__(self, predictions: torch.Tensor,
+                 gold_labels: torch.Tensor,
+                 mask: Optional[torch.Tensor]):
+
+        raise NotImplementedError
+
+    def get_metric(self, reset: bool) -> Union[float, Tuple[float, ...]]:
+        """
+        Compute and return the metric. Optionally also call :func:`self.reset`.
+        """
+        raise NotImplementedError
+
+    def reset(self) -> None:
+        """
+        Reset any accumulators or internal state.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def from_params(cls, params: Params):
+        metric_type = params.pop_choice("type", cls.list_available())
+        return cls.by_name(metric_type)(**params.as_dict())  # type: ignore
+
+
+@Metric.register("categorical_accuracy")
+class CategoricalAccuracy(Metric):
+    """
+    Categorical Top-K accuracy. Assumes integer labels, with
```

Comment:
I'd note in here that this is returned as a value from 0-100, not as a value from 0-1.